osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -442,5 +449,53 @@ public void Dispose()
         }
 
         #endregion
+
+        /// <summary>
+        /// Import a beatmap directly from game resources.
+        /// </summary>
+        /// <remarks>
+        /// Will mark as protected.
+        /// </remarks>
+        /// <param name="beatmapFile"></param>
+        /// <param name="beatmapHash"></param>
+        /// <returns></returns>
+        public WorkingBeatmap? ImportFromGameResources(string beatmapFile, string beatmapHash)
+        {
+            WorkingBeatmap? beatmap = null;
+
+            // First check if the beatmap is already imported and in a good state.
+            var setInfo = QueryBeatmapSet(b => b.Protected && b.Hash == beatmapHash);
+
+            if (setInfo != null)
+            {
+                setInfo.PerformRead(s =>
+                {
+                    if (s.Beatmaps.Count == 0)
+                        return;
+
+                    beatmap = GetWorkingBeatmap(s.Beatmaps.First());
+                });
+
+                if (beatmap != null)
+                    return beatmap;
+            }
+
+            // Intro beatmaps are generally made using the osu! ruleset.
+            // It might not be present in test projects for other rulesets.
+            // To avoid import errors / failures, skip import if we don't have an osu! ruleset.
+            if (rulesets.GetRuleset(0) == null)
+                return null;
+
+            // If we detect that the theme track or beatmap is unavailable this is either first startup or things are in a bad state.
+            // This could happen if a user has nuked their files store. for now, reimport to repair this.
+            var import = Import(new ZipArchiveReader(gameResources.GetStream(beatmapFile), Path.GetFileName(beatmapFile))).GetResultSafely();
+
+            if (import == null)
+                return null;
+
+            import.PerformWrite(b => b.Protected = true);
+
+            return GetWorkingBeatmap(import.Value.Beatmaps.First());
```

Comment:
As per test failures, this can't work at the usage site because `.Value` is being accessed from a load thread. The following passes tests at least:

```suggestion
            return GetWorkingBeatmap(import.PerformRead(set => set.Beatmaps.First().Detach()));
```